### PR TITLE
FEATURE: In slow mode dialog make "Enabled Until" mandatory

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**

### CODE REVIEW

**Review by eviltrout — APPROVED** [review]

File: app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js [hunk 1]
```diff
@@ -66,9 +66,10 @@ export default Controller.extend(ModalFunctionality, {
     }
   },
 
-  @discourseComputed("hours", "minutes", "seconds")
-  submitDisabled(hours, minutes, seconds) {
-    return this.saveDisabled || !(hours || minutes || seconds);
+  @discourseComputed("hours", "minutes", "seconds", "enabledUntil")
+  submitDisabled(hours, minutes, seconds, enabledUntil) {
+    const durationIsSet = hours || minutes || seconds;
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js:71]: maybe you could have a computed property durationIsSet this way we avoid having so many arguments here, more than 3 is kinda a code smell

File: app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ export default Controller.extend(ModalFunctionality, {
   hours: null,
   minutes: null,
   seconds: null,
-  saveDisabled: false,
+  submittingDisabled: false,
```
Comment (AndrewPrigorshnev) [inline app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js:16]: @jjaffeux I'm not sure about this naming, does this look good to you? 

The reason I've changed it is that there is no Save button on the modal dialog. This property is used to
disable submitting during a request to the server.

**Comment by AndrewPrigorshnev** [comment]
@jjaffeux Sorry I've reverted the last commit. It makes the problem with naming a bit worse. I propose to leave the current naming now even though it's not ideal.

I want to provide later a little fix of the behavior of this form and I will fix naming at the same time.

File: app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js (hunk 1 above)
Comment (AndrewPrigorshnev) [reply app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js]: I've fixed this.

File: app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js [hunk 3]
```diff
@@ -67,8 +67,13 @@ export default Controller.extend(ModalFunctionality, {
   },
 
   @discourseComputed("hours", "minutes", "seconds")
-  submitDisabled(hours, minutes, seconds) {
-    return this.saveDisabled || !(hours || minutes || seconds);
+  durationIsSet(hours, minutes, seconds) {
+    return hours || minutes || seconds;
+  },
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js:72]: great, now I think you can use `import { or } from "@ember/object/computed";``

```
durationIsSet: or("hours", "minutes", "seconds"),
```

I think it should work

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**